mac

```diff
@@ -190,6 +190,14 @@ if [ "$brew_last_change" -le "1471233600" ]; then
   print_done
 fi
 
+# Fix permission issue on /usr/local on OS 10.12
+if [ ! -w /usr/local ]; then
```

Comment:
Based on my experience `/usr/local` was not writeable which prevented `/usr/local/n` from being created. Homebrew advice seems to be to make only certain subdirectories writeable, and perhaps if it's doing this automatically it's missing `n`, but in that case this should still apply the fix as Homebrew doesn't seem to take the approach of chowning `/usr/local` itself.

So let's start with this, and only add a special case for certain subdirs if necessary.